src/reanimated2/hook/useScrollViewOffset.ts

```diff
@@ -89,15 +90,23 @@ function useScrollViewOffsetNative(
 
   useEffect(() => {
     // We need to make sure that listener for old animatedRef value is removed
-    if (scrollRef.current !== null) {
-      eventHandler.workletEventHandler.unregisterFromEvents();
+    if (scrollRef.current !== null && scrollRefTag.current !== null) {
+      eventHandler.workletEventHandler.unregisterFromEvents(
+        scrollRefTag.current
+      );
     }
+
+    // Store the ref and viewTag for future cleanup
     scrollRef.current = animatedRef.current;
+    scrollRefTag.current = animatedRef.getTag();
 
-    const viewTag = animatedRef.getTag();
-    eventHandler.workletEventHandler.registerForEvents(viewTag);
+    eventHandler.workletEventHandler.registerForEvents(scrollRefTag.current);
```

Comment:
This will crash if `animated.getTag()` will return a null. You can copy a guard from:

- #5862
 
to make it safer.